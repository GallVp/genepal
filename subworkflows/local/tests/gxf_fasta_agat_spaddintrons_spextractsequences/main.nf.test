nextflow_workflow {

    name "Test Subworkflow GXF_FASTA_AGAT_SPADDINTRONS_SPEXTRACTSEQUENCES"
    script "../../../gallvp/gxf_fasta_agat_spaddintrons_spextractsequences/main.nf"
    workflow "GXF_FASTA_AGAT_SPADDINTRONS_SPEXTRACTSEQUENCES"
    config './nextflow.config'

    tag "subworkflows"
    tag "subworkflows_gallvp"
    tag "subworkflows/gxf_fasta_agat_spaddintrons_spextractsequences"
    tag "modules/nf-core/gunzip"
    tag "agat/spextractsequences"
    tag "agat/spaddintrons"

    test("scaffold_4 - fasta - gff3") {

        when {
            workflow {
                """
                input[0] = Channel.of ( [
                    [ id:'test' ],
                    file("$baseDir" + '/subworkflows/local/tests/gxf_fasta_agat_spaddintrons_spextractsequences/testdata/scaffold_4.gff', checkIfExists: true)
                ] )
                input[1] = Channel.of ( [
                    [ id:'test' ],
                    file("$baseDir" + '/subworkflows/local/tests/gxf_fasta_agat_spaddintrons_spextractsequences/testdata/scaffold_4.fasta', checkIfExists: true)
                ] )
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }
}
