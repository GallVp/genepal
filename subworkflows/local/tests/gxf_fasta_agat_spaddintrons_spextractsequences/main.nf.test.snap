{
    "scaffold_4 - fasta - gff3": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.motifs.tsv:md5,b0a99a7a7bf598bbf1e25ca3690567c6"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.marked.gff3:md5,f464b2e0dc24238e69c3e9e1419c1b3b"
                    ]
                ],
                "2": [
                    "versions.yml:md5,1befbe41bc7abbf49767b8dc68877bc7",
                    "versions.yml:md5,65042e008b2466984150cb219a05291c"
                ],
                "marked_gff3": [
                    [
                        {
                            "id": "test"
                        },
                        "test.marked.gff3:md5,f464b2e0dc24238e69c3e9e1419c1b3b"
                    ]
                ],
                "motifs_tsv": [
                    [
                        {
                            "id": "test"
                        },
                        "test.motifs.tsv:md5,b0a99a7a7bf598bbf1e25ca3690567c6"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1befbe41bc7abbf49767b8dc68877bc7",
                    "versions.yml:md5,65042e008b2466984150cb219a05291c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-20T12:00:17.861955"
    }
}